internal/steplib/steplib.go

```diff
@@ -0,0 +1,117 @@
+package steplib
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/bitrise-io/stepman/models"
+	"github.com/lszucs/build-aggregator/internal/bitrise"
+	"github.com/lszucs/build-aggregator/internal/webhook"
+)
+
+// FetchSteps loads the step specs found at the configured path
+func FetchSteps(steplibSpecURL string) ([]models.StepModel, error) {
+	resp, err := http.Get(steplibSpecURL)
+	if err != nil {
+		return []models.StepModel{}, fmt.Errorf("getting step spec.json: %s", err)
+	}
+
+	if resp.StatusCode != 200 {
+		return []models.StepModel{}, fmt.Errorf("getting step spec.json: %s", resp.Status)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return []models.StepModel{}, fmt.Errorf("reading spec.json: %s", err)
+	}
+
+	var spec models.StepCollectionModel
+	if err := json.Unmarshal(body, &spec); err != nil {
+		return []models.StepModel{}, fmt.Errorf("deserializing spec.json %s: %s", body, err)
+	}
+
+	latests := []models.StepModel{}
+	for _, step := range spec.Steps {
+		latests = append(latests, getLatestVersion(step))
+	}
+
+	return latests, nil
+}
+
+func getLatestVersion(step models.StepGroupModel) models.StepModel {
```

Comment:
models.StepGroupModel holds the latestVersion already: https://github.com/bitrise-io/stepman/blob/master/models/models.go#L125

Reply:
In case of a full spec file, yes, but in case of the slim, it doesn't, unfortunately. I had a corresponding logic to support a full spec file too, but I decided to remove it in favour of simpler code and support the slim json only.